src/main/kotlin/com/terraformation/backend/accelerator/api/ScoringController.kt

```diff
@@ -0,0 +1,74 @@
+package com.terraformation.backend.accelerator.api
+
+import com.terraformation.backend.api.AcceleratorEndpoint
+import com.terraformation.backend.api.ApiResponse200
+import com.terraformation.backend.api.ApiResponse404
+import com.terraformation.backend.api.SuccessResponsePayload
+import com.terraformation.backend.db.accelerator.CohortPhase
+import com.terraformation.backend.db.accelerator.ScoreCategory
+import com.terraformation.backend.db.default_schema.ProjectId
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Schema
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@AcceleratorEndpoint
+@RequestMapping("/api/v1/accelerator/projects/{projectId}/scores")
+@RestController
+class ProjectScoringController() {
+  @ApiResponse200
+  @ApiResponse404
+  @GetMapping
+  @Operation(summary = "Gets vote selections for a single project.")
+  fun getProjectScores(
+      @PathVariable("projectId") projectId: ProjectId,
+  ): GetProjectScoresResponsePayload {
+    return GetProjectScoresResponsePayload(projectId, "testProject", emptyList())
+  }
+
+  @ApiResponse200
+  @ApiResponse404
+  @PutMapping
+  @Operation(
+      summary = "Upserts vote selections for a single project.",
+      description =
+          "Update the scores for the project phase. If the (project, phase, category) does not " +
+              "exist, a new entry is created. Setting a `score` to `null` removes the score.")
+  fun upsertProjectScores(
+      @PathVariable("projectId") projectId: ProjectId,
+      @RequestBody payload: UpsertProjectScoresRequestPayload,
+  ): UpsertProjectScoresResponsePayload {
+    return UpsertProjectScoresResponsePayload(projectId, payload.phase, emptyList())
+  }
+}
+
+data class Score(
```

Comment:
I think we will also need something like `inputType`, each `category` contains a number score (dropdown with -2 -> 2) and qualitative (text) input. For example "Carbon Score" and "Carbon Qualitative".

We also will have a (and possibly more in the future) system generated score. I think it would be alongside the other scores, so maybe something like `type: 'user' | 'system'` should be added. Perhaps @sgrimm has an opinion on where the system generated score will exist. We can also just do the math on the frontend, but I suspect folks outside engineering will want this value to be in the database.

Reply:
Qualitative is my oversight. Adding in the next patch. 

As for system generated score, are they going to be tied to score categories? We will need to update our table schema to reflect which category is user-updated and which is system generated. 

Will update API schema to accept int | String for score as well.

Reply:
I am thinking that we can keep every score with the following fields:

category
value: int?
qualitative: string?
modifiedTime

Each phase should always have all the categories at least according to the current PRD.

Reply:
I don't have a super strong opinion about how it is implemented, it seems like it should just be considered "another score" for simplicity. In the mock data I use `category: "calculated"` to differentiate, but it would probably be redundant if we additionally have `type: "system"`. 

Perhaps the `category` is nullable and only applies to user supplied scores? Alternatively we ditch `type` altogether and just use the `category` to differentiate between system and user values, IE every category that isn't "system" or "calculated" is a user value.

Reply:
> I am thinking that we can keep every score with the following fields:
> 
> category value: int? qualitative: string? modifiedTime
> 
> Each phase should always have all the categories at least according to the current PRD.

This looks good to me! The `modifiedTime` is only needed at the scorecard or phase scores level though.

Reply:
We will have `modifiedTime` per category because we have modifiedTime stored in each of the score row. We can either have the backend returned the latest modifiedTime, or have the frontend take the latestTime of the cards. Whichever one you prefer!

Reply:
@nickgraz I added a separate return for system scores. For now there may just be one that is hard-coded to be average. I think in the future, if there are more system scores, we will add a separate ENUM table and keep track of those. Will be nicer going forward to make the distinction between user score and system score. Let me know what you think!

Reply:
One slight snag with this approach, the "system calculated score" shouldn't be generated until all the other scores have a value. This is probably not a big deal, the math can be done on the fly if the conditions are right.

Another thing, which I think you should ask product about perhaps, this seems like the value would not be available in the database, only in the request flow. I _think_ they want it in the database for reporting purposes.

Reply:
We probably want the logic to be part of SearchService regardless. We either want to handle updating the calculated score on the database on writes, or have the service compute the score on request. 

Our database looks different depending on the approach but we should be able to expose that for reporting either way.

The first approach gives us easier access to the values (if we ever dump scores directly from the db for example)

The second approach gives us less coupling to worry about in the data perspective. 

I will make sure to check with product about whether we have that calculated score stored in database requirement.

Reply:
@sgrimm hoping to get your thoughts on this. Whether we should write calculated scores to the database, and keep it as a service output. We can make this decision a bit downstream as well.

Reply:
My inclination would be to store it in the database, though for a different reason: if we change the score calculation in the future, we may not want the calculated scores for older projects to change retroactively.

Reply:
Also, in terms of the API representation, the total score could be a single `Double?` field that's a sibling of the list of per-category scores.

Reply:
That sounds good to me. I think the system score changes will likely be part of a different PR. If the api data types look good to @nickgraz and @sgrimm, we can merge this one in!